<?xml version="1.0"?>
<launch>
 <rosparam file="$(find tangy_description)/controllers/eff_controllers.yaml" command="load"/>

    <include file="$(find tangerine_2dnav)/launch/my_robot_configuration.launch"/>
    
    <include file="$(find tangerine_2dnav)/launch/move_base.launch" />
<!--
    <include file="$(find openni2_launch)/launch/openni2.launch">
      <arg name="camera" value="bottom_camera"/>
      <arg name="device_id" value="#1"/>
    </include>
-->

    <include file="$(find bingo_detection)/launch/BingoDetection_server.launch"/>

    <node name="arm_player" pkg="drrobot_h20_arm_player" type="arm_player_eff" respawn="true"/>
    
    <node name="ArmMovementServer" pkg="drrobot_h20_arm_player" type="ArmMovement" respawn="true"/>

<!--    <node name="fault_diagnosis_core" pkg="fault_diagnosis" type="fault_diagnosis_core" respawn="true" output="screen"/>
    <node name="fault_monitor_webcam" pkg="fault_diagnosis" type="fault_monitor_webcam" respawn="true" output="screen"/>
    <node name="fault_monitor_rangefinder" pkg="fault_diagnosis" type="fault_monitor_rangefinder" respawn="true" output="screen"/>
    <node name="fault_monitor_axiscamera" pkg="fault_diagnosis" type="fault_monitor_axiscamera" respawn="true" output="screen"/>
    <node name="fault_monitor_environment" pkg="fault_diagnosis" type="fault_monitor_environment" respawn="true" output="screen"/>
    <node name="fault_monitor_disconnect" pkg="fault_diagnosis" type="fault_monitor_disconnect" respawn="true" output="screen"/>-->
    
    <include file="$(find moveit_tangy)/launch/move_group.launch">
      <arg name="publish_monitored_planning_scene" value="true" />
    </include>

     <!-- robot Gui Server -->
    <include file="$(find robot_gui)/launch/robot_gui.launch"/>
    
    <node pkg="bingo_game" type="bingo_game" name="bingo_game" output="screen" respawn="true"/>
     		<param name="resultFile" type="string" value="$(find bingo_game)/savedState/completedNumbers.txt" />
    		<param name="bingoNumbers" type="string" value="$(find bingo_game)/savedState/bingoNumbers.txt" />
    		<param name="introFile" type="string" value="$(find bingo_game)/talking/intro.txt" />
		<param name="demoIntroFile" type="string" value="$(find bingo_game)/talking/demo_intro.txt" />
    		<param name="jokeFile" type="string" value="$(find bingo_game)/talking/jokes.txt" />
    		<param name="outroFile" type="string" value="$(find bingo_game)/talking/outro.txt" />
    		<param name="smallTalkFile" type="string" value="$(find bingo_game)/talking/smallTalk.txt" />
    		<param name="progressFile" type="string" value="$(find bingo_game)/talking/progress.txt" />
    		<param name="helpTextFile" type="string" value="$(find bingo_game)/talking/help_text.txt" />
		<param name="demoHelloFile" type="string" value="$(find bingo_game)/talking/demo_hello.txt" />
        <param name="demoprogressFile" type="string" value="$(find bingo_game)/talking/progress.txt"/>
    		
    <!--<node name="help_light" pkg="help_indicators" type="help_light" respawn="true"/> -->
		
    <include file="$(find face_detection)/launch/FaceDetectionServerAXIS.launch"/>
    
    <node pkg="music_player" type="music_player" name="music_player" output="screen" respawn="true"/>
     		<param name="musicFile1" type="string" value="$(find music_player)/music/file1.mp3" />
     		<param name="musicFile2" type="string" value="$(find music_player)/music/file2.mp3" />
     		<param name="musicFile3" type="string" value="$(find music_player)/music/file3.mp3" />
     		<param name="musicFile4" type="string" value="$(find music_player)/music/file4.mp3" />
     		<param name="musicFile5" type="string" value="$(find music_player)/music/file5.mp3" />
     		<param name="musicFile6" type="string" value="$(find music_player)/music/file6.mp3" />
     		<param name="celebrationMusic" type="string" value="$(find music_player)/music/celebrate.mp3" />

    <node pkg="bingo_game" type="face_tracker" name="face_tracker" respawn="true"/>
    <!-- tts -->
    <node pkg="text_to_speech" name="text_to_speech" type="text_to_speech" output="screen" />

</launch>
